parustiko/src/lib.rs

```diff
@@ -1,4 +1,65 @@
 mod errors;
 mod protocol;
+mod version_exchange;
 
-pub fn runner() {}
+use crypto::encryption::aes::{GenericArray, AES};
+use crypto::encryption::Encryption;
+use errors::VersionExchangeError;
+use protocol::DecodeRaw;
+use protocol::{BinaryProtocolPacket, Decode};
+use std::io::{Read, Write};
+use std::net::TcpStream;
+use std::time::Duration;
+use version_exchange::{KeyExchange, SshVersion};
+
+// Read bytes from stream until CR and LF ('\r\n') occur
+fn read_header<T: Read>(arr: &mut T) -> Result<Vec<u8>, VersionExchangeError> {
+    let mut header = Vec::with_capacity(51);
+    let mut prev_byte = None;
```

Comment:
initialize the first byte with the first one from stream, this Option<> seems unnecessary